### PR TITLE
workaround to make enter key submit action-form properly

### PR DESCRIPTION
**Pull request by joncasdam**
JIRA:
https://vikingco.atlassian.net/browse/MVBE-6037

### CODE REVIEW

File: advanced_reports/static/advanced_reports/js/advreport.js [hunk 1]
```diff
@@ -77,6 +77,16 @@ $(function(){
                     instance.update_checkboxes();
                 });
 
+                // workaround to fix the behaviour of enter key with modal forms.
+                // with this the form should properly be submitted
+                $('body').on('keypress keyup', '.mbox_content input', function(e){
+                    var code = e.keyCode || e.which;
+                    if (code == 13) {
```
Comment (pieterdd) [inline advanced_reports/static/advanced_reports/js/advreport.js]: Does this cover both enter keys?

**Comment by pieterdd** [comment]
Have you verified that this does not wreak havoc in other Advanced Reports? I can imagine some reports having custom enter behavior.

File: advanced_reports/static/advanced_reports/js/advreport.js [hunk 2]
```diff
@@ -77,6 +77,16 @@ $(function(){
                     instance.update_checkboxes();
                 });
 
+                // workaround to fix the behaviour of enter key with modal forms.
+                // with this the form should properly be submitted
+                $('body').on('keypress keyup', '.mbox_content input', function(e){
```
Comment (pieterdd) [inline advanced_reports/static/advanced_reports/js/advreport.js]: Would it be better to hook onto a form submission event instead?

**Comment by pieterdd** [comment]
Can you increment the library version?

File: advanced_reports/static/advanced_reports/js/advreport.js (hunk 1 above)
Comment (joncasdam) [inline advanced_reports/static/advanced_reports/js/advreport.js]: yes, cover both.

File: advanced_reports/static/advanced_reports/js/advreport.js (hunk 2 above)
Comment (joncasdam) [inline advanced_reports/static/advanced_reports/js/advreport.js]: I got to this with Ingo and we have agreed that, actually, would be better a project to overview js for this repo. That's also why I've putted as 'workaround'

**Comment by joncasdam** [comment]
@pieterdd version number incremented

File: advanced_reports/static/advanced_reports/js/advreport.js (hunk 2 above)
Comment (pieterdd) [inline advanced_reports/static/advanced_reports/js/advreport.js]: Also, keep in mind that people outside of VikingCo also use this library. We don't want to break other people's workflow, either.

Comment (joncasdam) [inline advanced_reports/static/advanced_reports/js/advreport.js]: @pieterdd updated

File: advanced_reports/static/advanced_reports/js/advreport.js [hunk 3]
```diff
@@ -77,6 +77,18 @@ $(function(){
                     instance.update_checkboxes();
                 });
 
+                // workaround to fix the behaviour of enter key with modal forms.
+                // with this the form should properly be submitted and it was
+                // made to be as generic as possible
+                $('body').on('keypress keyup', '.mbox_content input', function(e){
+                    var code = e.keyCode || e.which;
+                    var parent_element = $(this).parents('#mbox, .mbox');
```
Comment (pieterdd) [inline advanced_reports/static/advanced_reports/js/advreport.js]: Why do you need both selectors?

Comment (joncasdam) [inline advanced_reports/static/advanced_reports/js/advreport.js]: because, since it's not standardized yet, 'mbox' sometimes it's used as an id or a class

Comment (pieterdd) [inline advanced_reports/static/advanced_reports/js/advreport.js]: Okay. I'd add this consideration to the inline docs.

**Comment by joncasdam** [comment]
@pieterdd comment added explaining the approach on the jquery selector

**Comment by pieterdd** [comment]
If retested manually, it's okay for me.

**Comment by Duologic** [comment]
Has been merged, just bump the version number and did a git commit --amend on it, apparently this has the side effect that it doesn't close the pull request.

**Pull request closed without merging**